rooms.js

```diff
@@ -789,6 +789,34 @@ class GlobalRoom {
 	modlog(text) {
 		this.modlogStream.write('[' + (new Date().toJSON()) + '] ' + text + '\n');
 	}
+	startLockdown(err, slow) {
+		if (this.lockdown) return;
+		let devRoom = Rooms('development');
+		Rooms.rooms.forEach((curRoom, id) => {
+			if (id === 'global') return;
+			if (err) {
+				let stack = Chat.escapeHTML(err.stack).split("\n").slice(0, 2).join("<br />");
```

Comment:
This needs to be moved out of the loop, since it doesn't need anything that changes per-iteration from inside the loop.